test/probes/pg.test.js

```diff
@@ -1,17 +1,423 @@
+/* global it, describe, before, after */
 'use strict'
 
-const semver = require('semver')
-const pkg = require('pg/package.json')
-
-//
-// so why are the files called pg6-minus and pg6-plus but
-// the test is for 7.0.0? because in theory v6 has all v7
-// features in it so could be tested against that both. we're
-// not doing that now to avoid the complications of interpreting
-// the results.
-//
-if (semver.lt(pkg.version, '7.0.0')) {
-  require('./pg6-minus')
-} else {
-  require('./pg6-plus')
+const helper = require('../helper')
+const { ao } = require('../1.test-common')
+
+const conf = ao.probes.pg
+
+const pkg = require('pg/package')
+const postgres = require('pg')
+
+const env = process.env
+const addr = helper.Address.from(env.AO_TEST_POSTGRES || 'postgres:5432')[0]
+// using a null password is valid.
+const password = 'AO_TEST_POSTGRES_PASSWORD' in env ? env.AO_TEST_POSTGRES_PASSWORD : 'xyzzy'
+
+// make a unique table name so multiple tests can run concurrently without colliding.
+const tName = 'tbl' + (env.AO_IX ? env.AO_IX : '')
+
+const extend = Object.assign
+
+const auth = {
+  host: addr.host,
+  port: addr.port,
+  user: env.AO_TEST_POSTGRES_USERNAME || 'postgres',
+  password: password,
+  database: 'test'
+}
+
+let hasNative = false
+let nativeVer = '0.0.0'
+
+try {
+  require('pg/lib/native')
+  nativeVer = require('pg-native/package').version
+  hasNative = true
+} catch (e) {
+  ao.loggers.test.info('test/probes/pg 6+ failed to load pg native')
+  hasNative = false
 }
+
+describe(`probes.pg6+ ${pkg.version} pg-native ${nativeVer}`, function () {
```

Comment:
nit: call it pg7+ ?

Reply:
better still - just call it `pg`. c1c53e7